src/core/solutions/brown_function.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+/* 
+ * File:   brown_function.cpp
+ * Author: Alejandro Ruiz Madruga <amadruga at estudiantes.uci.cu>
+ *
+ * Created on 28 de septiembre de 2019, 21:56
+ */
+
+#include <math.h>
+#include <core/solutions/brown_function.h>
+
+
+namespace dnn_opt
+{
+namespace core
+{
+namespace solutions
+{
+    brown_function* brown_function::make(generator* generator, unsigned int size)
+    {
+        auto* result = new brown_function(generator, size);
+        result->init();
+        return result;
+    }
+    float brown_function::calculate_fitness()
+    {
+        float result=0;
+        float* params=get_params();
+        
+        solution::calculate_fitness();
+        
+        for(int i=0;i<size()-1;i++)
+        {
+           result+=(pow(params[i],2*pow(params[i+1],2)+2)+pow(pow(params[i+1],2),pow(params[i],2)+1));

```

Comment:
Spaces between operators. I'm not sure if this long line is less than 80 characters (this is the standard). Is it possible to create auxiliary variables for calculations? 
It seems that `params[i+1],2)` can be stored in advance...
Also, I suggest using 2.0f for floats: something like ` 2*pow(params[i+1],2)` may give a few surprises

Reply:
What is 2.0f? You mean that i should create a variable with this operation and then multiplique them for 2

Reply:
`2.0f ` is the literal for float, `2`  is an integer, hence sometimes the compiler performs integer multiplication instead of float multiplication. That can lead to wrong results in calculations. So, I would suggest to modify this:

 `result+=(pow(params[i],2*pow(params[i+1],2)+2)+pow(pow(params[i+1],2),pow(params[i],2)+1));`

to this:

` result += (pow(params[i], 2.0f * pow(params[i + 1], 2.0f ) + 2.0f) + pow(pow(params[i + 1], 2.0f), pow(params[i], 2.0f) + 1.0f));`